core/hash/crc.odin

```diff
@@ -1,14 +1,14 @@
 package hash
 
-crc32 :: proc(data: []byte) -> u32 #no_bounds_check {
-	result := ~u32(0);
+crc32 :: proc(data: []byte, seed := u32(0)) -> u32 #no_bounds_check {
```

Comment:
Should the default seed be `~u32(0)` instead?

Reply:
No, the seed that is passed in, 0 by default, gets inverted to become the ~0 if run as before.
This way, to extend the hash you don't have to invert the previous hash first.

Reply:
I'm using the the CRC in the PNG reader without passing the optional parameter, and the Adler32 with the optional parameter as a rolling hash as I output bytes. They're correct, just more convenient.

Old CRC32 init:
`result := u32(~0);`

New CRC32 init:
```
seed := 0;
result := u32(~seed);
```

They're equivalent. When the optional seed for a preceding CRC run is not passed, the function will sub in the default 0 and it'll be inverted. The optimizer will elide that ~ and load result with ~0 directly when no seed is passed, I'm sure.

Doing it this way means that chained hashing doesn't require inverting the previous hash when passing it:

 ```
 crc := crc32(data);
 crc  = crc32(more_data, crc);
 
 versus:
 
 crc := crc32(data);
 crc  = crc32(more_data, ~crc);
 ```